src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java

```diff
@@ -269,6 +277,216 @@ private Optional<Pocket> findPocketForReload(ReloadBalanceRequest reloadRequest,
+        if (isRevertOperationAlreadyProvided) {
+            log.error("Revert operation for UUID: {} - already provided", revertRequest.getUuidToRevert());
+            throw new BusinessException("error.revert.operation.already.provided", "Revert operation already provided");
+        }
+
+        OperationType previousOperationType = operationToRevert.getOperationType();
+        log.info("Operation type to revert: {}", previousOperationType);
+        if (previousOperationType.equals(RELOAD)) {
+            return revertReloadOperation(balance, revertRequest, operationToRevert);
+        } else if (previousOperationType.equals(CHARGING)) {
+            return revertChargingOperation(balance, revertRequest, operationToRevert);
+        } else {
+            log.error("Could not operation provide, because revert operation type: {} - not supported", operationToRevert.getOperationType());
+            throw new BusinessException("error.revert.operation.type.unknown", "Revert operation type not supported");
+        }
+    }
+
+    private BalanceChangeEventDto revertReloadOperation(Balance balance,
+                                                        RevertBalanceOperationRequest revertRequest,
+                                                        BalanceChangeEventDto reloadOperationToRevert) {
+        String operationUuid = StringUtils.isNotBlank(revertRequest.getUuid())
+            ? revertRequest.getUuid()
+            : UUID.randomUUID().toString();
+
+        BalanceChangeEvent changeEvent = createBalanceChangeEvent(
+            operationUuid,
+            reloadOperationToRevert.getAmountTotal(),
+            Metadata.of(revertRequest.getMetadata()),
+            now(clock),
+            balance,
+            true,
+            REVERT_RELOAD
+        );
+        changeEvent.setRevertOperationId(revertRequest.getUuidToRevert());
+
+        log.info("Revert reload mode: {}", revertRequest.getRevertReloadMode());
+        if (revertRequest.getRevertReloadMode() == null) {
+            log.error("Could not operation provide, because empty revert reload mode");
+            throw new BusinessException("error.revert.reload.mode.empty", "Empty revert reload mode");
+        } else if (revertRequest.getRevertReloadMode().equals(EXCEPTION) || revertRequest.getRevertReloadMode().equals(AS_MANY_AS_POSSIBLE_FROM_POCKETS)) {
+            revertReloadByMode(revertRequest.getRevertReloadMode(), reloadOperationToRevert.getPocketChangeEvents(), changeEvent);
+        } else if (revertRequest.getRevertReloadMode().equals(AS_MANY_AS_POSSIBLE_FROM_BALANCE)) {
+            chargingPockets(balance, reloadOperationToRevert.getAmountTotal(), changeEvent, now(clock), true, null);
+        } else {
+            log.error("Could not operation provide, because unknown revert reload mode: {}", revertRequest.getRevertReloadMode());
```

Comment:
`Could not operation provide, because unknown revert reload mode: {}` -> `Could not perform operation, unknown revert reload mode: {}`

Reply:
Fixed